fix: reworked Toast system

nickmandev opened the pull request:
Implemented an EventBus system that will allow us to communicate between components without coupling them

Commit:
fix: reworked Toast system

Implemented an EventBus system that will allow us to communicate between components without coupling them

Signed-off-by: Nikolay Manchev <nikolay.manchev@limechain.tech>

ivalime reviewed (changes requested):
Fix ReviewProvider.tsx & Marketplace.tsx

Commit:
fix: components Martketplace and ReviewProvider were using setToast

These components were still using the old toast system

Signed-off-by: Nikolay Manchev <nikolay.manchev@limechain.tech>

Commit:
chore: fixing failing tests for components

Components: PolicyProvider, ReviewProvider, vulticonnectWalletService and Wallet

Signed-off-by: Nikolay Manchev <nikolay.manchev@limechain.tech>
